### PR TITLE
Increase version number allowed for System.IdentityModel.Tokens.Jwt in ServiceBus

### PR DESCRIPTION
**Pull request by raghu017**
By increasing the upper bound we can use version 6.5.0 of System.IdentityModel.Tokens.Jwt

### CODE REVIEW

**Comment by raghu017** [comment]
https://github.com/Azure/azure-sdk-for-net/issues/10872

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <AssemblyTitle>Azure ServiceBus SDK</AssemblyTitle>
     <Description>This is the next generation Azure Service Bus .NET Standard client library that focuses on queues &amp; topics. For more information about Service Bus, see https://azure.microsoft.com/en-us/services/service-bus/</Description>
-    <Version>4.2.0-preview.1</Version>
+    <Version>4.2.1</Version>
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj]: Please revert this; versioning in this repository follows our release process.  The version should not be adjusted outside of that.

**Comment by jsquire** [comment]
Hi @raghu017.   Thank you for your contribution.   I'm going to defer to @nemakam for an authoritative review and decision, as he is the subject matter expert for Service Bus.   

I would like to mention that, as a general rule, we strongly prefer not to override version compatibility at the level of a single service.  Version compatibility for dependencies is something that can have wide-reaching impact across the Azure libraries and bring subtle issues - especially when using the libraries for different services together in a single application..  

As a result, it is something that we need to be careful with and ensure proper test rigor.  Would you be so kind as to help us understand your process in testing these changes, especially with respect to scenarios where multiple Azure libraries are used together in the same project?

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj (hunk 1 above)
Comment (raghu017) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/Microsoft.Azure.ServiceBus.csproj]: Reverted this change. Will the versioning change from preview to release?

**Comment by raghu017** [comment]
> Hi @raghu017. Thank you for your contribution. I'm going to defer to @nemakam for an authoritative review and decision, as he is the subject matter expert for Service Bus.
> 
> I would like to mention that, as a general rule, we strongly prefer not to override version compatibility at the level of a single service without a compelling scenario to do so. Version compatibility for dependencies is something that can have wide-reaching impact across the Azure libraries and bring subtle issues - especially when using the libraries for different services together in a single application..
> 
> As a result, it is something that we need to be careful with and ensure proper test rigor. Would you be so kind as to help us understand your process in testing these changes, especially with respect to scenarios where multiple Azure libraries are used together in the same project?

For tests, all unit and end to end tests passed when they were run locally.

**Comment by nemakam** [comment]
@jsquire , another team had verified that it worked for them when they started using 6.x version of the library (on a private build that was given to them). 
Regarding the override at library level, how do you want to go about deciding whether this should be done at service level, or overall azure sdk level?

**Review by nemakam — APPROVED** [review]

**Comment by jsquire** [comment]
> @jsquire , another team had verified that it worked for them when they started using 6.x version of the library (on a private build that was given to them).
> Regarding the override at library level, how do you want to go about deciding whether this should be done at service level, or overall azure sdk level?

That's a larger question that we'll probably want to pull @mikeharder and @weshaggard in for discussion.  My primary concern in this case is not so much around "does it work for Service Bus" but more "what happens when you use Service Bus in an application with Storage, Event Hubs, Key Vault, and others?"    Event making this change at the service level introduces a bit of risk, though it may very well be warranted.   I'll admit to not having deep knowledge of the commitment around backwards compatibility and breaking changes for the JWT library, so it may turn out that I'm being overly conservative.

**Comment by nemakam** [comment]
Here are the release notes of the breaking changes - https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/releases

**Comment by raghu017** [comment]
Can we merge this PR?

**Comment by nemakam** [comment]
@raghu017 , waiting for @mikeharder  @weshaggard for their inputs. 
If we don't have an answer by monday, I'll merge this and we can try to get a new release.

**Comment by weshaggard** [comment]
There are a couple of issues I'd like to consider:

1) We should update this version in https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Packages.Data.props#L88 instead of at the individual project level. Based on a quick search (https://github.com/Azure/azure-sdk-for-net/search?l=XML&q=System.IdentityModel.Tokens.Jwt) the only 2 projects using not currently overriding this version are Microsoft.Azure.ServiceBus and Azure.Messaging.ServiceBus which we should ideally keep in sync.

2) We should update the version to be to a min version instead of a range so I'd suggest we change the version to "5.4.0" which enables any version of the library to be used that is 5.4.0 or higher. We generally avoid using package version ranges unless there is a known issue with higher versions that we need to cap it off at a higher version. This will avoid needing to do similar PRs in the future.

**Comment by jsquire** [comment]
> There are a couple of issues I'd like to consider:
> 
> 1. We should update this version in https://github.com/Azure/azure-sdk-for-net/blob/master/eng/Packages.Data.props#L88 instead of at the individual project level. Based on a quick search (https://github.com/Azure/azure-sdk-for-net/search?l=XML&q=System.IdentityModel.Tokens.Jwt) the only 2 projects using not currently overriding this version are Microsoft.Azure.ServiceBus and Azure.Messaging.ServiceBus which we should ideally keep in sync.

`Azure.Messaging.ServiceBus` should not have a reference here, as it is built on `Azure.Identity`;  I'm assuming this was a mistake when setting up the new library.   I'll request that it be removed.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**